gstools/field/base.py

```diff
@@ -107,12 +108,33 @@ def mesh(
                     self.model.dim, direction
                 )
             )
-        if hasattr(mesh, "centroids_flat"):
+        has_pyvista = False
+        try:
+            import pyvista as pv
+            has_pyvista = True
+        except:
+            pass
+        # convert pyvista mesh
+        if has_pyvista and pv.is_pyvista_dataset(mesh):
+            if points == "centroids":
+                pnts = mesh.cell_centers().points.T[select]
+            else:
+                pnts = mesh.points.T[select]
+            out = self.unstructured(pos=pnts, **kwargs)
+            # Deal with the output
+            if isinstance(out, np.ndarray):
+                mesh[name] = out
+            else:
+                # if multiple values are returned, take the first one
+                mesh[name] = out[0]
```

Comment:
This is really messy... we should leverage #60 to handle adding all returned arrays to the PyVista and meshio meshes

Reply:
This can now be discussed in #65

Reply:
This is probably the best as is and we can use #60 to change this

Reply:
@banesullivan I added a workaround for this problem.